rajawali/src/main/java/org/rajawali3d/materials/shaders/AShaderBase.java

```diff
@@ -294,31 +294,31 @@ public ShaderVar xy()
 		
 		public ShaderVar x()
 		{
-			ShaderVar v = getReturnTypeForOperation(mDataType, mDataType);
+			ShaderVar v = new RFloat();
 			v.setName(this.mName + ".x");
 			v.mInitialized = true;
 			return v;
 		}
 		
 		public ShaderVar y()
 		{
-			ShaderVar v = getReturnTypeForOperation(mDataType, mDataType);
+			ShaderVar v = new RFloat();
 			v.setName(this.mName + ".y");
 			v.mInitialized = true;
 			return v;
 		}
 		
 		public ShaderVar s()
 		{
-			ShaderVar v = getReturnTypeForOperation(mDataType, mDataType);
```

Comment:
I am confused, aside from float, what other data types would have a meaningful assign to x?